### ISSUE TITLE
TypeError: URL is not a constructor

### ISSUE THREAD

**Issue by Oleksii14** [kenigbolo/network-speed#96]
I receive this error while trying to run checkDownloadSpeed

My code:

```
const assetUrl = '...'
  const sizeInBytes = 8000;

  const speedInMb = (
    await new NetworkSpeed().checkDownloadSpeed(assetUrl, sizeInBytes)
  ).mbps;
```

**Comment by wakingweb** [comment]
I was getting this error as well, ended up changing line 3 in app.js from:
`const URL = require('url').URL;`
to
`const URL = require('url').Url;`

After some digging and help, looks like this is related to [defunctzombie/node-url#37](https://github.com/defunctzombie/node-url/issues/37)

Ideally, the dependency on URL could be removed to work properly in browser environments.

**Comment by kenigbolo** [comment]
@wakingweb @Oleksii14  I've updated the package with a patch for this based on @wakingweb suggestion. The caveat however is that when using on clientside, it'll require HTTPS strictly (which IMO is should suffice) however, it'll continue to work on node same way as it previously did.

Updated the code sandbox link here https://codesandbox.io/s/red-rgb-lx1wr to show this in principle. I'll close this issue now, but thanks for letting me know about this, and sorry it took so long - new dad problems :)
